exposed-java-time/src/test/kotlin/org/jetbrains/exposed/DefaultsTest.kt

```diff
@@ -555,4 +560,38 @@ class DefaultsTest : DatabaseTestsBase() {
             assertEquals(0, statements.size)
         }
     }
+
+    object DefaultTimestampTable : IntIdTable("test_table") {
+        val timestamp: Column<OffsetDateTime> =
+            timestampWithTimeZone("timestamp").defaultExpression(dbTimestampNow)
+    }
+
+    class DefaultTimestampEntity(id: EntityID<Int>) : Entity<Int>(id) {
+        companion object : EntityClass<Int, DefaultTimestampEntity>(DefaultTimestampTable)
+
+        var timestamp: OffsetDateTime by DefaultTimestampTable.timestamp
+    }
+
+    @Test
+    fun testCustomDefaultTimestampFunctionWithEntity() {
+        withTables(excludeSettings = TestDB.ALL - TestDB.ALL_POSTGRES - TestDB.MYSQL_V8 - TestDB.ALL_H2, DefaultTimestampTable) {
+            val entity = DefaultTimestampEntity.new {}
+
+            val timestamp = DefaultTimestampTable.selectAll().first()[DefaultTimestampTable.timestamp]
+
+            assertEquals(timestamp, entity.timestamp)
+        }
+    }
+
+    @Test
+    fun testCustomDefaultTimestampFunctionWithInsertStatement() {
+        // Only Postgres allows to get timestamp values directly from the insert statement due to implicit 'returning *'
+        withTables(excludeSettings = TestDB.ALL - TestDB.ALL_POSTGRES, DefaultTimestampTable) {
+            val entity = DefaultTimestampTable.insert { }
+
+            val timestamp = DefaultTimestampTable.selectAll().first()[DefaultTimestampTable.timestamp]
+
+            assertEquals(timestamp, entity[DefaultTimestampTable.timestamp])
+        }
+    }
```

Comment:
What do you think of adding these tests to the `exposed-kotlin-datetime` and `exposed-jodatime` modules too?